src/framework/core/js/ModelTransformationTransforms.js

```diff
@@ -409,15 +433,19 @@ var fluid = fluid || fluid_2_0_0;
         }
 
         var outputArr = [];
-        fluid.each(options, function (arrVal, inPath) {
-            var val = fluid.model.transform.getValue(inPath, undefined, transformer);
+        fluid.each(input, function (val, key) {
             if (val === transformSpec.presentValue) {
-                outputArr.push(arrVal);
+                outputArr.push(options[key]);
```

Comment:
So thinking through it, it seems to me that we would want the original behavior, ie., that the following transform:

```
```
{
    "detections": {
        "hasMouse": "supported",
        "hasKeyboard": "supported",
        "hasTrackpad": "not supported",
        "hasHeadtracker": "not supported",
        "hasTelephone": "supported"
    }
}
```